dockstore-common/src/test/java/io/dockstore/common/PipHelperTest.java

```diff
@@ -10,9 +10,15 @@
 public class PipHelperTest {
     @Test
     public void convertSemVerToAvailableVersion() {
-        Assert.assertEquals("1.10.0", PipHelper.convertSemVerToAvailableVersion(PipHelper.DEV_SEM_VER));
-        Assert.assertEquals("1.10.0", PipHelper.convertSemVerToAvailableVersion(null));
-        Assert.assertEquals("1.10.0", PipHelper.convertSemVerToAvailableVersion("9000.9000.9000"));
+        Assert.assertEquals("1.13.0", PipHelper.convertSemVerToAvailableVersion(PipHelper.DEV_SEM_VER));
+        Assert.assertEquals("1.13.0", PipHelper.convertSemVerToAvailableVersion(null));
+        Assert.assertEquals("1.13.0", PipHelper.convertSemVerToAvailableVersion("9000.9000.9000"));
+        Assert.assertEquals("1.13.0", PipHelper.convertSemVerToAvailableVersion("1.12.1"));
```

Comment:
I was following the current pattern where cwltool updates don't track the releases; and the test was to make sure that when we create a major, minor or hotfix (patch) release after 1.12.0 that the code will point to the requirements3.txt file which updates cwltool. If we don't want to update cwltool until the next major release then maybe the piphelper.java code should be

```
       if (semVer.greaterThanOrEqualTo(Version.valueOf("1.13.0"))) {
            return "1.13.0";
        }
```

Reply:
Hmmm, might be a regression https://ucsc-cgl.atlassian.net/browse/DOCK-669

Reply:
FYI, there was another fix in the area, #3989 (may or may not be relevant to this discussion -- just seeing the thread reminded me of that bug I had created).

Reply:
From what I can tell, the criteria for selecting the cwltool requirements file is based on the webservice API version retrieved in the UI [here](https://github.com/dockstore/dockstore-ui2/blob/688de436321c60163c055f3b6825f5be9943ca7d/src/app/loginComponents/onboarding/downloadcliclient/downloadcliclient.component.ts#L40). 

This is used in the URL [here](https://github.com/dockstore/dockstore-ui2/blob/688de436321c60163c055f3b6825f5be9943ca7d/src/app/loginComponents/onboarding/downloadcliclient/downloadcliclient.component.ts#L180) in the curl command to download the cwltool requirements file.

The curl command results in a call to the webservice [getRunnerDependencies](https://github.com/dockstore/dockstore/blob/2e2f53ec0c0ca82b994c234e79d547ef3a7ddcf8/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java#L276), which then calls convertSemVerToAvailableVersion. The contents of the requirements file is returned.

To get the cwltool version the api version is passed [here](https://github.com/dockstore/dockstore-ui2/blob/688de436321c60163c055f3b6825f5be9943ca7d/src/app/loginComponents/onboarding/downloadcliclient/downloadcliclient.component.ts#L43) and the cwltool version is extracted from returned requirements file)

Reply:
Ok, but big picture. 
This doesn't really make sense as written if I understand correctly.  Since this is going into develop, this will not affect anyone until we release 1.13. Then when we release 1.13, users of old 1.12 hotfix versions will suddenly get dependencies for 1.13. 

> Additionally the [URL to download the CLI](https://github.com/dockstore/dockstore-ui2/blob/688de436321c60163c055f3b6825f5be9943ca7d/src/app/loginComponents/onboarding/downloadcliclient/downloadcliclient.component.ts#L41) should really point to https://github.com/dockstore/dockstore-cli and not https://github.com/dockstore/dockstore

Makes sense to me, can you follow up with an issue?

Reply:
https://ucsc-cgl.atlassian.net/browse/SEAB-4238

Reply:
now using 
       if (semVer.greaterThanOrEqualTo(Version.valueOf("1.13.0"))) {
            return "1.13.0";
        }